lib/new_relic/agent/agent_helpers/special_startup.rb

```diff
@@ -10,7 +10,7 @@ module SpecialStartup
         # requests, we need to wait until the children are forked
         # before connecting, otherwise the parent process sends useless data
         def using_forking_dispatcher?
-          if [:puma, :passenger, :unicorn].include?(Agent.config[:dispatcher])
+          if [:puma, :passenger, :unicorn, :falcon].include?(Agent.config[:dispatcher])
```

Comment:
Falcon was added to this list because of its ability to fork. Falcon can also run in a threaded mode instead of a forking one, but given that:

> There is no guarantee that your process would be forked or threaded.

https://github.com/socketry/falcon/issues/53#issuecomment-461979683

It makes sense to always defer when Falcon is in play.